File: 排序/快速排序/quick_sort.py
def quickSort(arr):
  if len(arr) <= 1:
    return arr
  else:
    pivot = arr[0]
    less = []
    greater = []
    for i in arr[1:]:
      if i <= pivot:
        less.append(i)
      else:
        greater.append(i)
    return quickSort(less) + [pivot] + quickSort(greater)

arr = [6, 2, 1, 8, 7, 3, 5]
print('result: {}'.format(quickSort(arr)))

File: 排序/快速排序/quick_sort.js
function quickSort(arr) {
  if (arr.length <= 1) {
    return arr;
  } else {
    const pivot = arr[0];
    const less = [];
    const greater = [];
    for (let i = 1; i < arr.length; i++) {
      const el = arr[i];
      if (el <= pivot) {
        less.push(el);
      } else {
        greater.push(el);
      }
    }
    return quickSort(less)
      .concat(pivot)
      .concat(quickSort(greater));
  }
}

const arr = [6, 2, 1, 8, 7, 3, 5];
console.log("result: ", quickSort(arr));
